mapping/collection.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+module Mapping
+  class AdminSet
+    require 'csv'
+    require 'yaml'
+
+    extend Extensions::BasicValueHandler
+
+    ##
+    # Convert data in a csv file to hash in YAML format
+    #
+    # @param[String] value - the path to the csv file
+    # return - [Hash] of YAML
+    def csv_to_yaml(value)
+      data = CSV.read (value, :headers => true).map(&:to_hash)
+      data.to_yaml
+    end
+
+  ##
+  # Get ownerId from object_properties and parse DSpace owning collection name from lookup table 
+  def owner_collection_name
+    lookup = csv_to_yaml('../../lookup/collectionlist.csv')
+    collection_list = lookup.select { |l| l[:collection_handle] == object_properties['ownerId'] }
+    [ collection_list[:collection_name] ]
+  end 
+
+  ##
+  # Get ownerId from object_properties and parse DSpace mapped collection name(s) from lookup table
+  def other_collection_name
```

Comment:
This needs the proper method signature, with plural I think because it is going to return an array of strings.

`def other_collection_names(value, *args)`